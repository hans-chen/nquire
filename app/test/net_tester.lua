#!/usr/bin/lua

package.path = "../?.lua;./?.lua"
 
require "tester"
require "net"

th = Tester.new()

local function test_get_interface_ip()
	local ip, err = net.get_interface_ip("eth0")
	print("ip(eth0)=" .. (ip or "nil") .. ", err=" .. (err or "nil"))
	local ip, err = net.get_interface_ip("none")
	print("ip(none)=" .. (ip or "nil") .. ", err=" .. (err or "nil"))
end

local function test_get_interface_mac()
	local mac, err = net.get_interface_mac("eth0")
	print("mac(eth0)=" .. (mac or "nil") .. ", err=" .. (err or "nil"))
	local mac, err = net.get_interface_mac("wlan0")
	print("mac(wlan0)=" .. (mac or "nil") .. ", err=" .. (err or "nil"))
	local mac, err = net.get_interface_mac("none")
	print("mac(none)=" .. (mac or "nil") .. ", err=" .. (err or "nil"))
end

th:run( test_get_interface_ip, "test_get_interface_ip" )
th:run( test_get_interface_mac, "test_get_interface_mac" )
