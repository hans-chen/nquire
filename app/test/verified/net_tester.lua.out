# *** Running module/net_tester.lua


# *** Start test 'test_get_interface_ip'
ip(eth0)=192.168.1.5, err=nil
ip(none)=nil, err=No such device

# *** Start test 'test_get_interface_mac'
mac(eth0)=64:31:50:6e:a0:8c, err=nil
mac(wlan0)=00:13:e8:b4:e0:b3, err=nil
mac(none)=nil, err=Error requesting mac-address

# *** Start test 'test_gethostbyname_test'
1 = 	0.0.0.1
1.2 = 	1.0.0.2
1.2.3 = 	1.2.0.3
1.2.3.4 = 	1.2.3.4
1.2.3.4.5= Error: No address associated with hostname
localhost = 	127.0.0.1
www.vdtai.nl = 	85.17.131.4
www.vdtai.nl/index.html= Error: No address associated with hostname
www.newland-id.nl = 	194.150.208.116
www.unknown42.nl= Error: No address associated with hostname
